providers/TanstackQueryProvider.tsx

```diff
@@ -0,0 +1,16 @@
+"use client";
+
+import { useState } from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+
+export default function TanstackQueryProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [queryClient] = useState(() => new QueryClient());
```

Comment:
Why useState? just use `const queryClient = new QueryClient();`
```suggestion
  const queryClient = new QueryClient();
```